client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/util/FileUtils.java

```diff
@@ -152,4 +162,47 @@ public static void replaceDir(Consumer<Path> dirContentUpdater, Path target, Pat
     FileUtils.mkdirs(target.getParent());
     FileUtils.moveDir(work, target);
   }
+  
+  /**
+   * On Windows, retries the provided IO operation a number of times, in an effort to make the operation succeed.
+   * 
+   * Operations that might fail on Windows are file & directory move, as well as file deletion. These failures 
+   * are typically caused by the virus scanner and/or the Windows Indexing Service. These services tend to open a file handle 
+   * on newly created files in an effort to scan their content.
+   * 
+   * @param runnable the runnable whose execution should be retried
+   */
+  private static void retry(IORunnable runnable) throws IOException {
+    for (int retries = isWindows()? 30: 1; retries > 0; retries--) {
```

Comment:
isn't 30 a little excessive? It would take 3seconds to fail, which is a long time in SonarLint environment.

Reply:
We don't expect it to fail. *We expect it so succeed*. Moreover, it would succeed on the second or third retry at most which is *100ms wait time or 200ms at most*. Always, or almost always. But if - for whatever reason - once in a lifetime - the PC is under a heavy load, and McAfee is stressed to complete its virus checking on time, we don't want to report failure. The user would then wait for 3 seconds. Not a big deal IMHO.

I think if you want to continue arguing over the same lines, you should concentrate on the Thread.sleep timeout, rather than on the number of retry, as at least one Thread.sleep call will always happen on Windows, if McAfee is installed and screwing things for us.

Reply:
Alright then.
Just a note: here you are assuming that when an `AccessDeniedException` happens, it's a temporary situation. However, I imagine this happening due to an actual lack of permissions and in this case we want to fail as fast as possible, hence my comment. In this situation what matters is the total timeout, not the number retries or the duration of each sleep.

Reply:
Right... yet: what is the chance the user not to have permissions in - say - the temporary space where these files reside.

Moreover, if there is a general permissions' issue, I don't think you can recover from that. In that case, most likely the stack will be unrolled and a general error (maybe even internal error?) will appear in Eclipse, VS etc. The fact that it will appear 3 seconds later is probably not the end of the world.